Tests/SourceKitLSPTests/ExecuteCommandTests.swift

```diff
@@ -136,6 +136,104 @@ final class ExecuteCommandTests: XCTestCase {
+        import SwiftSyntax
+        import SwiftSyntaxBuilder
+        import SwiftSyntaxMacros
+
+        public struct StringifyMacro: ExpressionMacro {
+          public static func expansion(
+            of node: some FreestandingMacroExpansionSyntax,
+            in context: some MacroExpansionContext
+          ) -> ExprSyntax {
+            guard let argument = node.argumentList.first?.expression else {
+              fatalError("compiler bug: the macro does not have any arguments")
+            }
+
+            return "(\(argument), \(literal: argument.description))"
+          }
+        }
+
+        @main
+        struct MyMacroPlugin: CompilerPlugin {
+          let providingMacros: [Macro.Type] = [
+            StringifyMacro.self,
+          ]
+        }
+        """#,
+        "MyMacroClient/MyMacroClient.swift": """
+        @freestanding(expression)
+        public macro stringify<T>(_ value: T) -> (T, String) = #externalMacro(module: "MyMacros", type: "StringifyMacro")
+
+        func test() {
+          1️⃣#2️⃣stringify3️⃣(1 + 2)
+        }
+        """,
+      ],
+      manifest: SwiftPMTestProject.macroPackageManifest
+    )
+    try await SwiftPMTestProject.build(at: project.scratchDirectory)
+
+    let (uri, positions) = try project.openDocument("MyMacroClient.swift")
+
+    let argsToBeTested = [
+      ExpandMacroCommand(
+        positionRange: positions["1️⃣"]..<positions["3️⃣"],
+        textDocument: TextDocumentIdentifier(uri)
+      ),
+      ExpandMacroCommand(
+        positionRange: positions["2️⃣"]..<positions["3️⃣"],
+        textDocument: TextDocumentIdentifier(uri)
+      ),
+    ]
+
+    for args in argsToBeTested {
+      let metadata = SourceKitLSPCommandMetadata(textDocument: TextDocumentIdentifier(uri))
+
+      var command = args.asCommand()
+      command.arguments?.append(metadata.encodeToLSPAny())
+
+      let request = ExecuteCommandRequest(command: command.command, arguments: command.arguments)
+
+      let expectation = XCTestExpectation(description: "Handle Show Document Request")
+      var showDocumentRequestURI: DocumentURI? = nil
+
+      project.testClient.handleSingleRequest { (req: ShowDocumentRequest) in
+        showDocumentRequestURI = req.uri
+        expectation.fulfill()
+        return ShowDocumentResponse(success: true)
+      }
+
+      let result = try await project.testClient.send(request)
+
+      guard let resultArray: [MacroExpansionEdit] = Array(fromLSPArray: result ?? .null) else {
+        XCTFail("Result is not an array.")
+        return
+      }
+
+      XCTAssertEqual(resultArray.count, 1)
+      XCTAssertEqual(resultArray.only?.newText, "(1 + 2, \"1 + 2\")")
+
+      await fulfillment(of: [expectation], timeout: 10.0)
+
+      let url = try XCTUnwrap(showDocumentRequestURI?.fileURL)
+
+      XCTAssert(
+        (try? url.checkResourceIsReachable()) ?? false
+      )
+
+      let fileContents = try XCTUnwrap(try? String(contentsOf: url, encoding: .utf8))
```

Comment:
I think you can save the `XCTUnwrap` dance here and just throw if it fails. The benefit would be that this also shows the error why the string couldn’t load the file contents when it fails.

```suggestion
      let fileContents = try String(contentsOf: url, encoding: .utf8)
```